clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java

```diff
@@ -68,7 +68,7 @@
  */
 public class ConfigDef {
 
-    public static final Object NO_DEFAULT_VALUE = new String("");
+    public static final Object NO_DEFAULT_VALUE = "";
```

Comment:
Thanks for the explanation.  Let me add a comment explaining why the `new` is there.